diff --git a/python/Makefile.am b/python/Makefile.am
index 0305bcc..b7d0e17 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -57,9 +57,9 @@ nodist_libvirtmod_la_SOURCES = libvirt.c libvirt.h
 libvirtmod_la_CFLAGS = $(WARN_CFLAGS) $(WARN_PYTHON_CFLAGS)
 
 libvirtmod_la_LDFLAGS = -module -avoid-version -shared -L$(top_builddir)/src/.libs \
-	$(CYGWIN_EXTRA_LDFLAGS)
+	$(CYGWIN_EXTRA_LDFLAGS) $(MINGW_EXTRA_LDFLAGS) -L/python/libs
 libvirtmod_la_LIBADD = $(mylibs) \
-	$(CYGWIN_EXTRA_LIBADD) $(CYGWIN_EXTRA_PYTHON_LIBADD)
+	$(CYGWIN_EXTRA_LIBADD) $(CYGWIN_EXTRA_PYTHON_LIBADD) -lpython26
 
 libvirtmod_qemu_la_SOURCES = libvirt-qemu-override.c typewrappers.c
 nodist_libvirtmod_qemu_la_SOURCES = libvirt-qemu.c libvirt-qemu.h
diff --git a/src/remote/remote_driver.c b/src/remote/remote_driver.c
index 031167a..8f16f05 100644
--- a/src/remote/remote_driver.c
+++ b/src/remote/remote_driver.c
@@ -151,6 +151,70 @@ static void remoteDomainEventQueue(struct private_data *priv, virDomainEventPtr
 /* Helper functions for remoteOpen. */
 static char *get_transport_from_scheme (char *scheme);
 
+#ifdef WIN32
+
+static char *remoteWin32CertPaths[5];
+
+static int
+remoteInitCertPaths(void)
+{
+    const char *appdata = getenv("APPDATA");
+
+    if (appdata == NULL || *appdata == '\0') {
+        appdata = "C:";
+
+        VIR_WARN("APPDATA not set, falling back to '%s'", appdata);
+    }
+
+    if (virAsprintf(&remoteWin32CertPaths[0],
+                    "%s\\libvirt\\pki\\CA\\cacert.pem", appdata) < 0 ||
+        virAsprintf(&remoteWin32CertPaths[1],
+                    "%s\\libvirt\\pki\\libvirt\\private\\clientkey.pem", appdata) < 0 ||
+        virAsprintf(&remoteWin32CertPaths[2],
+                    "%s\\libvirt\\pki\\libvirt\\clientcert.pem", appdata) < 0 ||
+        virAsprintf(&remoteWin32CertPaths[3],
+                    "%s\\libvirt\\pki\\libvirt\\private\\serverkey.pem", appdata) < 0 ||
+        virAsprintf(&remoteWin32CertPaths[4],
+                    "%s\\libvirt\\pki\\libvirt\\servercert.pem", appdata) < 0) {
+        virReportOOMError();
+        return -1;
+    }
+
+    return 0;
+}
+
+static const char *
+remoteCACertPath(void)
+{
+    return remoteWin32CertPaths[0];
+}
+
+static const char *
+remoteClientKeyPath(void)
+{
+    return remoteWin32CertPaths[1];
+}
+
+static const char *
+remoteClientCertPath(void)
+{
+    return remoteWin32CertPaths[2];
+}
+
+static const char *
+remoteServerKeyPath(void)
+{
+    return remoteWin32CertPaths[3];
+}
+
+static const char *
+remoteServerCertPath(void)
+{
+    return remoteWin32CertPaths[4];
+}
+
+#endif /* WIN32 */
+
 #ifdef WITH_LIBVIRTD
 static int
 remoteStartup(int privileged ATTRIBUTE_UNUSED)
@@ -5078,6 +5142,9 @@ remoteRegister (void)
 {
     remoteDriver = &remote_driver;
 
+    #ifdef WIN32
+        if (remoteInitCertPaths()) return -1;
+    #endif
     if (virRegisterDriver (&remote_driver) == -1) return -1;
     if (virRegisterNetworkDriver (&network_driver) == -1) return -1;
     if (virRegisterInterfaceDriver (&interface_driver) == -1) return -1;
diff --git a/src/remote/remote_driver.h b/src/remote/remote_driver.h
index 1504eec..4eccb3f 100644
--- a/src/remote/remote_driver.h
+++ b/src/remote/remote_driver.h
@@ -41,12 +41,19 @@ unsigned long remoteVersion(void);
 # define LIBVIRTD_CONFIGURATION_FILE SYSCONFDIR "/libvirtd.conf"
 
 /* Defaults for PKI directory. */
-# define LIBVIRT_PKI_DIR SYSCONFDIR "/pki"
-# define LIBVIRT_CACERT LIBVIRT_PKI_DIR "/CA/cacert.pem"
-# define LIBVIRT_CLIENTKEY LIBVIRT_PKI_DIR "/libvirt/private/clientkey.pem"
-# define LIBVIRT_CLIENTCERT LIBVIRT_PKI_DIR "/libvirt/clientcert.pem"
-# define LIBVIRT_SERVERKEY LIBVIRT_PKI_DIR "/libvirt/private/serverkey.pem"
-# define LIBVIRT_SERVERCERT LIBVIRT_PKI_DIR "/libvirt/servercert.pem"
-
+# ifndef WIN32
+#  define LIBVIRT_PKI_DIR SYSCONFDIR "/pki"
+#  define LIBVIRT_CACERT LIBVIRT_PKI_DIR "/CA/cacert.pem"
+#  define LIBVIRT_CLIENTKEY LIBVIRT_PKI_DIR "/libvirt/private/clientkey.pem"
+#  define LIBVIRT_CLIENTCERT LIBVIRT_PKI_DIR "/libvirt/clientcert.pem"
+#  define LIBVIRT_SERVERKEY LIBVIRT_PKI_DIR "/libvirt/private/serverkey.pem"
+#  define LIBVIRT_SERVERCERT LIBVIRT_PKI_DIR "/libvirt/servercert.pem"
+# else
+#  define LIBVIRT_CACERT remoteCACertPath()
+#  define LIBVIRT_CLIENTKEY remoteClientKeyPath()
+#  define LIBVIRT_CLIENTCERT remoteClientCertPath()
+#  define LIBVIRT_SERVERKEY remoteServerKeyPath()
+#  define LIBVIRT_SERVERCERT remoteServerCertPath()
+# endif
 
 #endif /* __VIR_REMOTE_INTERNAL_H__ */
