--- a/testThreads.c	2010-01-21 21:32:25 +0000
+++ b/testThreads.c	2010-01-21 21:33:15 +0000
@@ -107,7 +107,7 @@
 
 	for (i = 0; i < num_threads; i++) {
 	    results[i] = NULL;
-	    tid[i] = (pthread_t) -1;
+	    memset(&tid[i], 0, sizeof(pthread_t));
 	}
 
 	for (i = 0; i < num_threads; i++) {
--- a/python/Makefile.am	2009-09-24 15:32:00 +0000
+++ b/python/Makefile.am	2010-09-11 20:32:31 +0000
@@ -31,7 +31,7 @@
 
 libxml2mod_la_SOURCES = libxml.c types.c
 nodist_libxml2mod_la_SOURCES = libxml2-py.c
-libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@ @PYTHON_LIBS@
+libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@ -L/python/libs @PYTHON_LIBS@ -Wl,libxml2mod.def
 
 libxml.c: libxml2-py.h # to generate before to compile
 
--- a/python/Makefile.in	2010-09-10 22:50:20 +0000
+++ b/python/Makefile.in	2010-09-10 22:50:30 +0000
@@ -222,7 +222,7 @@
 PERL = @PERL@
 PYTHON = @PYTHON@
 PYTHON_INCLUDES = @PYTHON_INCLUDES@
-PYTHON_LIBS = @PYTHON_LIBS@
+PYTHON_LIBS = -L/python/libs @PYTHON_LIBS@
 PYTHON_SITE_PACKAGES = @PYTHON_SITE_PACKAGES@
 PYTHON_SUBDIR = @PYTHON_SUBDIR@
 PYTHON_TESTS = @PYTHON_TESTS@
@@ -388,7 +388,7 @@
 @WITH_PYTHON_TRUE@python_LTLIBRARIES = libxml2mod.la
 @WITH_PYTHON_TRUE@libxml2mod_la_SOURCES = libxml.c types.c
 @WITH_PYTHON_TRUE@nodist_libxml2mod_la_SOURCES = libxml2-py.c
-@WITH_PYTHON_TRUE@libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@ @PYTHON_LIBS@
+@WITH_PYTHON_TRUE@libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@ -L/python/libs @PYTHON_LIBS@ -Wl,libxml2mod.def
 @WITH_PYTHON_TRUE@python_DATA = \
 @WITH_PYTHON_TRUE@	libxml2.py
 
--- a/python/libxml2mod.def
+++ b/python/libxml2mod.def
@@ -0,0 +1,2 @@
+EXPORTS
+    initlibxml2mod
